lib/cretonne/src/ir/dfg.rs

```diff
@@ -657,6 +665,28 @@ impl DataFlowGraph {
                         })
     }
 
+    /// Removes the argument `val` from `ebb`. Returns true if the removal is sucessful.
+    pub fn remove_ebb_arg(&mut self, ebb: Ebb, val: Value) -> bool {
```

Comment:
Ok, I'll transform this into a proper method. 

* OK for the tests.
* Ok fro the value table.
* Ok for the `swap_remove` semantics, it works for the use in `ssa::SSABuilder` because we process ebb arguments in order and when we remove an arg from an ebb we haven't appended jump arguments corresponding to it or the last ebb argument in the branch instruction.